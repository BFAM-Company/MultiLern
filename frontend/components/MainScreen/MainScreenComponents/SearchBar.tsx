import React from 'react';
import { Animated, Dimensions, TextInput } from 'react-native';
import { View, Text, SafeAreaView, StyleSheet, ImageBackground, Image, TouchableOpacity, Platform } from 'react-native';




function SearchBar({pageSwitcher}: any) {
  return (
    <View style={styles.mainContainer}>
        <TextInput 
            style={styles.TextInput}
            placeholder='wyszukaj zadania...'
        />
        <View style={styles.iconContainer}>
            <Image 
                source={require('./../../../assets/search-icon.png')}
                style={styles.icon}
            />
        </View>
    </View>
  );
}

const styles = StyleSheet.create({
    mainContainer:{
        width:'80%',
        height:50,
        maxWidth:700,
        minWidth:300,
        display:'flex',
        flexDirection:'row',
        backgroundColor:'rgb(33,33,43)',
        borderRadius:100,
        shadowColor: "#000",
        shadowOffset: {
                width: 0,
                height: -100,
        },
        shadowOpacity: 0.44,
        shadowRadius: 10.32,

        elevation: 16,
    },
    iconContainer:{
        width:'20%',
        height:'100%',
        display:'flex',
        alignItems:'center',
        justifyContent:'center',
        backgroundColor:'rgb(33,33,43)',
        borderTopRightRadius:20,
        borderBottomRightRadius:20,

    },
    icon:{
        width:20,
        height:20,
        tintColor:'#fff'
    },
    TextInput:{
        width:'80%',
        height:'100%',
        backgroundColor:'#fff',
        paddingLeft:10,
        borderRadius:20,
    }
})


export default SearchBar;