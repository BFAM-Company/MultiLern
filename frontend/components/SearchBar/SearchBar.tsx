import React, { useEffect, useState } from 'react';
import { Pressable, TextInput, TouchableOpacity } from 'react-native';
import { View, StyleSheet, Image } from 'react-native';
import Button from '../Button/Button';

function SearchBar({currentText, pageSwitcher, setModalVisibility, modalVisibility}: any) {
  const [searchedText, setSearchedText] = useState<string>('')

  useEffect(()=>{
    setSearchedText(currentText)
  }, [currentText])
  return (
    <View style={styles.mainContainer}>
        <TextInput
            testID='searchExercise'
            style={styles.TextInput}
            placeholder='wyszukaj zadania...'
						defaultValue={searchedText}
						onChangeText={newText => setSearchedText(newText)}
        />
        <View style={styles.iconContainer}>
            <TouchableOpacity testID='search-icon' onPress={() => { if(modalVisibility) {setModalVisibility(false)};pageSwitcher('Excercises', {searchableText: searchedText})}}>
                <Image 
                    source={require('./../../assets/search-icon.png')}
                    style={styles.icon}
                />
            </TouchableOpacity>
        </View>
    </View>
  );
}

const styles = StyleSheet.create({
    mainContainer:{
        width:'80%',
        height:50,
        maxWidth:700,
        minWidth:300,
        display:'flex',
        flexDirection:'row',
        backgroundColor:'rgb(33,33,43)',
        borderRadius:100,
        shadowColor: "#000",
        shadowOffset: {
                width: 0,
                height: 0,
                // height: -100,
        },
        shadowOpacity: 0.44,
        shadowRadius: 10.32,

        elevation: 16,
    },
    iconContainer:{
        width:'20%',
        height:'100%',
        display:'flex',
        alignItems:'center',
        justifyContent:'center',
        backgroundColor:'rgb(33,33,43)',
        borderTopRightRadius:20,
        borderBottomRightRadius:20,

    },
    icon:{
        width:20,
        height:20,
        tintColor:'#fff'
    },
    TextInput:{
        width:'80%',
        height:'100%',
        backgroundColor:'#fff',
        paddingLeft:10,
        borderRadius:20,
    },
    // button: {
    //     backgroundColor: 'none',
    //     color: 'inherit',
    //     // border: 'none',
    //     elevation: 0,
    //     borderRadius: 0,
    //     padding: 0,
    //     // font: 'inherit',
    // }
})


export default SearchBar;