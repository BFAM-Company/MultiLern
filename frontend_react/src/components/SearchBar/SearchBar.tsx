import React, { useState, useEffect } from 'react';
import { useNavigate } from 'react-router-dom';

interface SearchBarProps {
  currentText: string;
  setModalVisibility: (visible: boolean) => void;
  modalVisibility: boolean;
}

function SearchBar({ currentText, setModalVisibility, modalVisibility }: SearchBarProps) {
  const [searchedText, setSearchedText] = useState<string>('');
  const navigate = useNavigate()
  useEffect(() => {
    setSearchedText(currentText);
  }, [currentText]);

  return (
    <div style={{
        width: '80%',
        height: 50,
        maxWidth: 700,
        minWidth: 300,
        display: 'flex',
        flexDirection: 'row',
        backgroundColor: 'rgb(33,33,43)',
        borderRadius: 100,
        boxShadow: '0 0 10px rgba(0,0,0,0.44)',}}>
      <input
        type="text"
        style={{width: '80%',
    height: '100%',
    backgroundColor: '#fff',
    paddingLeft: 10,
    borderRadius: 20,
    border: 'none',}}
        placeholder="wyszukaj zadania..."
        value={searchedText}
        onChange={(e) => setSearchedText(e.target.value)}
      />
      <div style={{
        width: '20%',
    height: '100%',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: 'rgb(33,33,43)',
    borderTopRightRadius: 20,
    borderBottomRightRadius: 20,
      }}>
        <button onClick={() => { if (modalVisibility) setModalVisibility(false); navigate(`/excercises?searchableText=${searchedText}`)}}>
          <img src={require('../../assets/search-icon.png')} alt="search" style={{width: 20,
    height: 20,
    color: '#fff',}} />
        </button>
      </div>
    </div>
  );
}


export default SearchBar;
