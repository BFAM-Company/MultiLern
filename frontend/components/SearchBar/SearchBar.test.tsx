import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react-native';
import SearchBar from './SearchBar';

describe('SearchBar', () => {
  afterEach(cleanup);
    it('renders correctly', () => {
        const mockPageSwitcher = jest.fn();
        const { toJSON } = render(<SearchBar pageSwitcher={mockPageSwitcher} />);
        expect(toJSON()).toMatchSnapshot();
    });

    it('updates searchedText state when typing in TextInput', () => {
        const mockPageSwitcher = jest.fn();
        const { getByPlaceholderText } = render(
            <SearchBar pageSwitcher={mockPageSwitcher} />
        );

        const textInput = getByPlaceholderText('wyszukaj zadania...');
        fireEvent.changeText(textInput, 'example search text');

        expect(textInput.props.defaultValue).toBe('example search text');
    });

    it('calls pageSwitcher function with correct parameters when icon is pressed', () => {
        const mockPageSwitcher = jest.fn();
        const mockSetModalVisibility = jest.fn();
        const { getByTestId } = render(
        <SearchBar
            pageSwitcher={mockPageSwitcher}
            setModalVisibility={mockSetModalVisibility}
            modalVisibility={true}
        />
        );

        const icon = getByTestId('search-icon');
        fireEvent.press(icon);

        expect(mockSetModalVisibility).toHaveBeenCalled();
        expect(mockPageSwitcher).toHaveBeenCalledWith('Excercises', {
        searchableText: undefined,
        });
    });

   it('does not call setModalVisibility function when modalVisibility is true and icon is pressed', () => {
        const mockPageSwitcher = jest.fn();
        const mockSetModalVisibility = jest.fn();
        const { getByTestId } = render(
            <SearchBar
            pageSwitcher={mockPageSwitcher}
            setModalVisibility={mockSetModalVisibility}
            modalVisibility={false}
            />
        );

        const icon = getByTestId('search-icon');
        fireEvent.press(icon);

        expect(mockSetModalVisibility).not.toHaveBeenCalled();
    });



});
