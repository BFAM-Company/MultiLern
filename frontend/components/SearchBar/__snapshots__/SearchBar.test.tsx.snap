// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "rgb(33,33,43)",
      "borderRadius": 100,
      "display": "flex",
      "elevation": 16,
      "flexDirection": "row",
      "height": 50,
      "maxWidth": 700,
      "minWidth": 300,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 0.44,
      "shadowRadius": 10.32,
      "width": "80%",
    }
  }
>
  <TextInput
    onChangeText={[Function]}
    placeholder="wyszukaj zadania..."
    style={
      {
        "backgroundColor": "#fff",
        "borderRadius": 20,
        "height": "100%",
        "paddingLeft": 10,
        "width": "80%",
      }
    }
    testID="searchExercise"
  />
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "rgb(33,33,43)",
        "borderBottomRightRadius": 20,
        "borderTopRightRadius": 20,
        "display": "flex",
        "height": "100%",
        "justifyContent": "center",
        "width": "20%",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
      testID="search-icon"
    >
      <Image
        source={1}
        style={
          {
            "height": 20,
            "tintColor": "#fff",
            "width": 20,
          }
        }
      />
    </View>
  </View>
</View>
`;
